src/controllers/things_controller.js

```diff
@@ -204,22 +229,31 @@ ThingsController.ws('/:thingId/', function(websocket, request) {
       }));
       return;
     }
-    if (request.messageType === Constants.SET_PROPERTY) {
-      let setRequests = Object.keys(request.data).map(property => {
-        let value = request.data[property];
-        return device.setProperty(property, value);
-      });
-      Promise.all(setRequests).catch(err => {
-        // If any set fails, send an error
-        websocket.send(JSON.stringify({
-          messageType: Constants.ERROR,
-          data: {
-            status: '400 Bad Request',
-            message: err,
-            request: request
-          }
-        }));
-      });
+
+    switch (request.messageType) {
+      case Constants.SET_PROPERTY: {
+        let setRequests = Object.keys(request.data).map(property => {
+          let value = request.data[property];
+          return device.setProperty(property, value);
+        });
```

Comment:
I don't quite understand what this is trying to do?
Is it just setting multiple properties at the same time for a single device?

Reply:
Yes, given a request of the form
```javascript
{
  messageType: 'setProperty',
  data: {
    on: true,
    hue: 120
  }
}
```
it will set both properties